### PR TITLE
feat(core): support npm workspaces in lock file utils

### PR DESCRIPTION
**Pull request by meeroslav**
> Note: The #13585 should be merged first as part of that PR is used here as well

## Related Issue(s)

Related to #9761

### CODE REVIEW

File: packages/nx/src/lock-file/__fixtures__/pnpm.lock.ts [hunk 1]
```diff
@@ -2878,44 +2878,28 @@ packages:
 `;
 
 /**
```
Comment (meeroslav) [inline packages/nx/src/lock-file/__fixtures__/pnpm.lock.ts:2880]: We are extracting workspace fixtures to  workspaces.lock.ts

File: packages/nx/src/lock-file/lock-file.spec.ts [hunk 2]
```diff
@@ -239,7 +239,7 @@ describe('lock-file', () => {
       const lockFileData = parseNpmLockFile(npmLockFileWithAliases);
       const lockFile = stringifyNpmLockFile(lockFileData);
 
-      expect(lockFile).toEqual(npmLockFileWithAliases);
```
Comment (meeroslav) [inline packages/nx/src/lock-file/lock-file.spec.ts:242]: NPM is fragile with props ordering so comparing JSON's makes more sense

File: packages/nx/src/lock-file/npm.ts [hunk 3]
```diff
@@ -177,10 +197,17 @@ function mapPackageDependency(
 
     mappedPackages[packageName][key] = {
       ...(value as Omit<PackageDependency, 'packageMeta'>),
-      ...(!value.integrity && {
-        actualVersion: value.version,
-        version: value.resolved,
-      }),
+      ...(!value.integrity &&
+        value.version && {
```
Comment (meeroslav) [inline packages/nx/src/lock-file/npm.ts:201]: This `&& value.version &&` is the only addition. The rest of the file changes come from https://github.com/nrwl/nx/pull/13585

File: packages/nx/src/lock-file/npm.ts [hunk 4]
```diff
@@ -197,10 +197,11 @@ function mapPackageDependency(
...
```
Comment (meeroslav) [inline packages/nx/src/lock-file/npm.ts:201]: workspace project has no `version` in lock file

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**